### PR TITLE
OMPL Constrained State Space Planning rebase+orientation

### PR DESCRIPTION
**Pull request by gautz**
### Description

Now that the PR implementing orientation representation with rotation vector [has been merged](https://github.com/ros-planning/moveit/pull/2402), we can now solve orientation constrained trajectories with OMPL. The orientation constraints are defined with bounds in a similar way that position constraints can be defined as a box in the pending PR for OMPL Constrained State Space Planning

Co-authored-by: @JeroenDM jeroendemaeyer@live.be

This PR is consists of:
- a rebase of #2273
- an extension to support orientation constraints
- Corresponding tutorial: https://github.com/ros-planning/moveit_tutorials/pull/651

Required/overlapping PR:
- https://github.com/ros-planning/moveit/pull/2273

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [x] Extend the tutorials / documentation https://github.com/ros-planning/moveit_tutorials/pull/651
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by gautz** [comment]
Some examples of orientation constraints with collision-avoidance:
- [1,0.3,pi] radians
https://user-images.githubusercontent.com/64833674/123644202-713aca80-d825-11eb-8807-eeab7d015e5e.mp4
- [0.3,pi,0.3] radians
https://user-images.githubusercontent.com/64833674/123644343-9596a700-d825-11eb-84aa-8dec2b2e6c7b.mp4
- [pi,0.3,0.3] radians
https://user-images.githubusercontent.com/64833674/123644469-b2cb7580-d825-11eb-8ae3-a4286658f5cb.mp4

File: moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp [hunk 1]
```diff
@@ -61,6 +61,7 @@ moveit::core::RobotState* ompl_interface::TSStateStorage::getStateStorage() cons
   if (it == thread_states_.end())
   {
     st = new moveit::core::RobotState(start_state_);
+    st->setToDefaultValues();  // avoid uninitialized memory
```
Comment (v4hn) [inline moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp:64]: Please see https://github.com/ros-planning/moveit/pull/2273#discussion_r631760679 .
In addition to removing this line, it's probably worth to port the new ROS2 test for it?

Comment (gautz) [reply moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp:64]: Which would be the cleanest git procedure to port the test? Cherry picking?

**Comment by v4hn** [comment]
You can try to cherry-pick, though the repositories diverged quite a bit by now and porting between them is not entirely trivial in many cases.
Personally I would be lazy, copy&paste the code and commit with `--author=original_author_name` if it applies (mostly) as-is in MoveIt1.

File: moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp (hunk 1 above)
Comment (gautz) [reply moveit_planners/ompl/ompl_interface/src/detail/threadsafe_state_storage.cpp]: done

**Comment by gautz** [comment]
Closed in favor of #2273 and a PR on orientation constraints to be opened

**Pull request closed without merging**